osu.Game/Overlays/Toolbar/Toolbar.cs

```diff
@@ -41,6 +41,8 @@ public class Toolbar : VisibilityContainer, IKeyBindingHandler<GlobalAction>
         // Toolbar and its components need keyboard input even when hidden.
         public override bool PropagateNonPositionalInputSubTree => true;
 
+        protected override bool Handle(UIEvent e) => e is MouseEvent;
```

Comment:
I don't like this line. Is there a reason we need to handle in this way particularly?

Specifically: I don't think this is going to work with touch event conversion, if I recall how that works. Also, this is done nowhere else in game code, so if it is done here I would expect multiple sentences explaining why it is required.

Reply:
My intention when suggesting that was to handle all positional UI events (while excluding non-positional), but this may not be the correct way to do so.

Reply:
And what's the issue with just using `OnMouseDown => true`? It looks to work as expected.

Reply:
I guess. If it doesn't break with drags then sure.

Reply:
This sounds like `Toolbar` should inherit from `OverlayContainer` instead, which has mostly similar logic but allows for sharing implementation and potentially handling touch events at that level that way.